Encounter Me.Api/Trail generation Utilities/GooglePoints.cs

```diff
@@ -0,0 +1,127 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Text;
+
+namespace Encounter_Me.Api.Trail_generation_Utilities
+{
+    /// <summary>
+    /// See https://developers.google.com/maps/documentation/utilities/polylinealgorithm
+    /// </summary>
+    public static class GooglePoints
+    {
+        /// <summary>
+        /// Decode google style polyline coordinates.
+        /// </summary>
+        /// <param name="encodedPoints"></param>
+        /// <returns></returns>
+        public static IEnumerable<Coordinate> Decode(string encodedPoints)
+        {
+            if (string.IsNullOrEmpty(encodedPoints))
+                throw new ArgumentNullException("encodedPoints");
+
+            char[] polylineChars = encodedPoints.ToCharArray();
+            int index = 0;
+
+            int currentLat = 0;
+            int currentLng = 0;
+            int next5bits;
+            int sum;
+            int shifter;
+
+            while (index < polylineChars.Length)
+            {
+                // calculate next latitude
+                sum = 0;
+                shifter = 0;
+                do
+                {
+                    next5bits = (int)polylineChars[index++] - 63;
+                    sum |= (next5bits & 31) << shifter;
+                    shifter += 5;
+                } while (next5bits >= 32 && index < polylineChars.Length);
```

Comment:
could you explain what is going on here? Seems like you are trying to somehow modify the `polylineChars` char array, but I can't understand what you are trying to achieve.

Reply:
Probably should have mentioned this in the script, but this is code from: https://gist.github.com/shinyzhu/4617989
Google trails API returns encoded polyline, which needs to be decoded into actual geojson to be used, this script does just that.
Seems like in this exact part algorithm decodes the encoded polyline attributes with bit shifting.